src/nupic/ntypes/ArrayBase.cpp

```diff
@@ -291,15 +252,16 @@ bool ArrayBase::has_buffer() const { return (buffer_.get() != nullptr); }
 void ArrayBase::convertInto(ArrayBase &a, size_t offset, size_t maxsize) const {
   if (maxsize == 0)
     maxsize = getCount() + offset;
-  if (maxsize > a.getMaxElementsCount()) {
+  if (maxsize > a.getCount()) {
     a.allocateBuffer(maxsize);
     a.zeroBuffer();
   }
-  if (offset == 0) {
-    // This could be the first buffer of a Fan-In set.
-    if (a.getCount() != maxsize)
-      a.setCount(maxsize);
-  }
+	// TODO:  Comment this out until we are sure that it is not needed.
+  //if (offset == 0) {
```

Comment:
what was this for?Why we don't use it but you wanna keep that?

Reply:
It was for handling Fan-in merge when buffers could be variable length (Sparse arrays).  I don't think any other code was depending on this logic being here so if nothing breaks I will remove the commented out code.